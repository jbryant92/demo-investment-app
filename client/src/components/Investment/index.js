import React, { Component } from 'react';
import {
  Query
} from 'react-apollo';
import {
  Segment,
  Dimmer,
  Loader,
  Container,
  Header,
  Button,
  Card,
  Icon
} from 'semantic-ui-react';
import { Row }  from 'reactstrap';

import query from '../../queries/getInvestment';
import CampaignCard from '../CampaignCard'
import '../../App.css'

class Investment extends Component {
  render() {
    const { id } = this.props.match.params

    return (
      <Query
        query={query}
        variables={{ id }}
      >
        {({ loading, error, data }) => {
          if (loading) return(
            <Dimmer active inverted>
              <Loader size='large'>Loading</Loader>
            </Dimmer>
          );

          if (error) return <p>{error}</p>;

          const { campaign, amount } = data.investment;
          return (
            <Container textAlign='center'>
              <Header as='h2' icon textAlign='center'>
                <Icon name='trophy' circular />
                <Header.Content>Investment Successful!</Header.Content>
                 <Header.Subheader>Congratulations! You invested Â£{ amount } in { campaign.name }.</Header.Subheader>
              </Header>

              <Card.Group centered={ true }>
                <CampaignCard campaign={ campaign } clickable={ false }/>
              </Card.Group>

              <Container>
                <Row className='input-row'>
                  <Button positive href="/">
                    Continue investing
                    <Icon name='right arrow'/>
                  </Button>
                </Row>

              </Container>
            </Container>
          );
        }}
      </Query>
    )
  }
}

export default Investment;
