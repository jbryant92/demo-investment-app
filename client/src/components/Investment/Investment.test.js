import React from 'react';
import ReactDOM from 'react-dom';
import Investment from './index';
import enzymeToJson from 'enzyme-to-json'
import { shallow } from 'enzyme'

it('renders without crashing', () => {
  const instance = shallow(
        <Investment match={{params: {id: 1}}} />
    )
    expect(enzymeToJson(instance)).toMatchSnapshot()
});
