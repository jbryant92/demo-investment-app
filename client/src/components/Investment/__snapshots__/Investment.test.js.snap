// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<Query
  query={
    Object {
      "definitions": Array [
        Object {
          "directives": Array [],
          "kind": "OperationDefinition",
          "name": Object {
            "kind": "Name",
            "value": "getInvestment",
          },
          "operation": "query",
          "selectionSet": Object {
            "kind": "SelectionSet",
            "selections": Array [
              Object {
                "alias": undefined,
                "arguments": Array [
                  Object {
                    "kind": "Argument",
                    "name": Object {
                      "kind": "Name",
                      "value": "id",
                    },
                    "value": Object {
                      "kind": "Variable",
                      "name": Object {
                        "kind": "Name",
                        "value": "id",
                      },
                    },
                  },
                ],
                "directives": Array [],
                "kind": "Field",
                "name": Object {
                  "kind": "Name",
                  "value": "investment",
                },
                "selectionSet": Object {
                  "kind": "SelectionSet",
                  "selections": Array [
                    Object {
                      "alias": undefined,
                      "arguments": Array [],
                      "directives": Array [],
                      "kind": "Field",
                      "name": Object {
                        "kind": "Name",
                        "value": "amount",
                      },
                      "selectionSet": undefined,
                    },
                    Object {
                      "alias": undefined,
                      "arguments": Array [],
                      "directives": Array [],
                      "kind": "Field",
                      "name": Object {
                        "kind": "Name",
                        "value": "campaign",
                      },
                      "selectionSet": Object {
                        "kind": "SelectionSet",
                        "selections": Array [
                          Object {
                            "alias": undefined,
                            "arguments": Array [],
                            "directives": Array [],
                            "kind": "Field",
                            "name": Object {
                              "kind": "Name",
                              "value": "id",
                            },
                            "selectionSet": undefined,
                          },
                          Object {
                            "alias": undefined,
                            "arguments": Array [],
                            "directives": Array [],
                            "kind": "Field",
                            "name": Object {
                              "kind": "Name",
                              "value": "name",
                            },
                            "selectionSet": undefined,
                          },
                          Object {
                            "alias": undefined,
                            "arguments": Array [],
                            "directives": Array [],
                            "kind": "Field",
                            "name": Object {
                              "kind": "Name",
                              "value": "sector",
                            },
                            "selectionSet": undefined,
                          },
                          Object {
                            "alias": undefined,
                            "arguments": Array [],
                            "directives": Array [],
                            "kind": "Field",
                            "name": Object {
                              "kind": "Name",
                              "value": "imageUrl",
                            },
                            "selectionSet": undefined,
                          },
                          Object {
                            "alias": undefined,
                            "arguments": Array [],
                            "directives": Array [],
                            "kind": "Field",
                            "name": Object {
                              "kind": "Name",
                              "value": "country",
                            },
                            "selectionSet": Object {
                              "kind": "SelectionSet",
                              "selections": Array [
                                Object {
                                  "alias": undefined,
                                  "arguments": Array [],
                                  "directives": Array [],
                                  "kind": "Field",
                                  "name": Object {
                                    "kind": "Name",
                                    "value": "name",
                                  },
                                  "selectionSet": undefined,
                                },
                              ],
                            },
                          },
                          Object {
                            "alias": undefined,
                            "arguments": Array [],
                            "directives": Array [],
                            "kind": "Field",
                            "name": Object {
                              "kind": "Name",
                              "value": "targetAmount",
                            },
                            "selectionSet": undefined,
                          },
                          Object {
                            "alias": undefined,
                            "arguments": Array [],
                            "directives": Array [],
                            "kind": "Field",
                            "name": Object {
                              "kind": "Name",
                              "value": "fundedAmount",
                            },
                            "selectionSet": undefined,
                          },
                          Object {
                            "alias": undefined,
                            "arguments": Array [],
                            "directives": Array [],
                            "kind": "Field",
                            "name": Object {
                              "kind": "Name",
                              "value": "percentageComplete",
                            },
                            "selectionSet": undefined,
                          },
                          Object {
                            "alias": undefined,
                            "arguments": Array [],
                            "directives": Array [],
                            "kind": "Field",
                            "name": Object {
                              "kind": "Name",
                              "value": "numberOfInvestors",
                            },
                            "selectionSet": undefined,
                          },
                        ],
                      },
                    },
                  ],
                },
              },
            ],
          },
          "variableDefinitions": Array [
            Object {
              "defaultValue": undefined,
              "directives": Array [],
              "kind": "VariableDefinition",
              "type": Object {
                "kind": "NonNullType",
                "type": Object {
                  "kind": "NamedType",
                  "name": Object {
                    "kind": "Name",
                    "value": "ID",
                  },
                },
              },
              "variable": Object {
                "kind": "Variable",
                "name": Object {
                  "kind": "Name",
                  "value": "id",
                },
              },
            },
          ],
        },
      ],
      "kind": "Document",
      "loc": Object {
        "end": 301,
        "start": 0,
      },
    }
  }
  variables={
    Object {
      "id": "1",
    }
  }
>
  <Component />
</Query>
`;
